xonsh/aliases.py

```diff
@@ -109,6 +114,22 @@ def source_bash(args, stdin=None):
     return source_foreign(args, stdin=stdin)
 
 
+def source_cmd(args, stdin=None):
+    """Simple cmd.exe-specific wrapper around source-foreign."""
+    args = list(args)
+    fpath = locate_binary(args[0])
+    args[0] = fpath if fpath else args[0]
+    args.insert(0, 'cmd')
+    args.append('--interactive=0')
+    args.append('--sourcer=call')
+    args.append('--funcscmd=echo.')
+    args.append('--aliascmd=echo.')
+    args.append('--envcmd=set')
+    args.append('--safe=0')
+
+    return source_foreign(args, stdin=stdin)
```

Comment:
@scopatz . Just wondered. How do I wrap an alias defined in this way in a function. I am wrapping the `source_cmd()` function further down (https://github.com/scopatz/xonsh/pull/790/files#diff-68932bc9bfe7673583b7958ee52eff5bR271 ) , where I need to preserve the `$PROMPT` var.